src/main/jbake/content/blog/2017/06/9/opentracing-spring-boot.adoc

```diff
@@ -0,0 +1,147 @@
+= OpenTracing Spring Boot Instrumentation
+Pavol Loffay
+2017-6-9
+:icons: font
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, apm, tracing, jaeger, opentracing
+
+In this demo series we are going to look at how simple it is to instrument various Java frameworks using
+http://opentracing.io[OpenTracing]. You will see that it requires minimal changes to the
+application code. In the last demo we will have microservice apps deployed on Kubernetes and
+all services will be traced with OpenTracing complaint tracing system.
+
+In this first demo we are going to develop and trace simple Spring Boot app.
+
+== Create a web application
+First let's write a simple web app, no let's generate it! All we have to do is just to select a web dependency.
+
+ifndef::env-github[]
+image::/img/blog/2017/spring-initializr.png[caption="Figure 1: ", title="Spring boot generator."]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/spring-initializr.png[caption="Figure 1: ", title="Spring boot generator."]
+endif::[]
+
+Now the application is generated, but it does not contain any web controller. We are going to implement a simple
+controller with two methods. One will return hello world string and the other creates HTTP request which calls
```

Comment:
return hello world string -> return a greeting.  (or 'return the string Hello World')